### PR TITLE
feat: add feedback loop functionality

### PR DESCRIPTION
**Pull request by gldgrnt**
Addressing the requirements in [this issue](https://github.com/wmcadigital/wmn-sprints/issues/22).

### CODE REVIEW

**Review by daylesalmon — DISMISSED** [review]
Fantastic work on this @gldgrnt! 🥳 

It looks good and seems to be working. Is this linked to the live API? (ready to be used elsewhere as is?)

File: src/wmnds/patterns/feedback-loop/_properties.njk.json [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      "arrayOptions": [
+        {
+          "name": "contentText",
+          "type": "string",
+          "description": "Text shown once 'Is this page useful?' feedback is submitted. If <code class='wmnds-website-inline-code'>contentHTML</code> is provided, this argument will be ignored."
+        },
+        {
+          "name": "contentHTML",
+          "type": "string",
+          "description": "HTMl shown once 'Is this page useful?' feedback is submitted. If <code class='wmnds-website-inline-code'>contentHTML</code> is provided, the <code class='wmnds-website-inline-code'>contentText</code> argument will be ignored."
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/feedback-loop/_properties.njk.json:31]:
```suggestion
        {
          "name": "contentHTML",
          "type": "string",
          "description": "HTML shown once 'Is this page useful?' feedback is submitted. If <code class='wmnds-website-inline-code'>contentHTML</code> is provided, the <code class='wmnds-website-inline-code'>contentText</code> argument will be ignored."
```

File: src/wmnds/patterns/footer/_footer.njk [hunk 2]
```diff
@@ -4,12 +4,46 @@
...
+          labelText: 'I consent to being contacted about the issue I am reporting (optional)'
+        },
+        {
+          type: 'checkbox',
+          name: 'privacy',
+          labelHTML: 'I agree to <a href="#">WMCA’s Privacy Policy</a>',
+          required: true
+        }
+      ]
     }) | trim | indent(4)
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/footer/_footer.njk:47]:
```suggestion
    })
```

File: src/wmnds/patterns/feedback-loop/_example.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  const errorMessageClass = 'wmnds-fe-error-message';
+
+  // Helper functions
+  const nodeListToArray = nodeList => Array.prototype.slice.call(nodeList);
+  const createErrorMessage = textContet => {
+    const element = document.createElement('span');
+    element.classList.add(errorMessageClass);
+    element.textContent = textContet;
+    return element;
+  };
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/feedback-loop/_example.js:22]:
```suggestion
  const createErrorMessage = textContent => {
    const element = document.createElement('span');
    element.classList.add(errorMessageClass);
    element.textContent = textContent;
    return element;
  };
```

File: src/wmnds/patterns/feedback-loop/_properties.njk.json [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        },
+        {
+          "name": "labelText",
+          "type": "string",
+          "description": "<strong>Required</strong> Text label to be used for the input. If <code class='wmnds-website-inline-code'>labelHTML</code> is provided, this argument will be ignored."
+        },
+        {
+          "name": "labelHTML",
+          "type": "string",
+          "description": "<strong>Required</strong> HTMl to be used in the input label. If <code class='wmnds-website-inline-code'>labelHTML</code> is provided, <code class='wmnds-website-inline-code'>labelText</code> will be ignored."
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/feedback-loop/_properties.njk.json:97]:
```suggestion
          "description": "<strong>Required</strong> HTML to be used in the input label. If <code class='wmnds-website-inline-code'>labelHTML</code> is provided, <code class='wmnds-website-inline-code'>labelText</code> will be ignored."
```

**Review by catiarodriguescosta — CHANGES_REQUESTED** [review]
Looks good! 👍 Just found some typos!

**Review by catiarodriguescosta — APPROVED** [review]
👍

**Review by daylesalmon — APPROVED** [review]

**Pull request merged**